ship_functions.py

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Aug 15 11:37:18 2019
+
+@author: j-bd
+
+"""
+
+import os
+import argparse
+import shutil
+
+from sklearn.model_selection import train_test_split
+import numpy as np
+import pandas as pd
+import skimage.io
+
+from mrcnn import utils
+from mrcnn import model
+from mrcnn import visualize
+import ship_config
+import ship_dataset
+
+
+def create_parser():
+    '''Get the informations from the operator'''
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-c", "--command", required=True,
+                        help="choice between 'train' and 'detection'")
+    parser.add_argument("-of", "--origin_folder", required=True,
+                        help="path to the Kaggle folder containing all data")
+    parser.add_argument("-pf", "--project_folder", required=True,
+                        help="path to your project folder")
+    parser.add_argument("-w", "--weights", required=True,
+                        help="Choice between 'coco', 'last', 'imagenet', or"\
+                        "the path to a weights file usedusable by mrcnn algorithm"\
+                        "to detect object")
+    parser.add_argument("-sr", "--split_rate", type=float, default=0.8,
+                        help="split rate between train and validation dataset during"\
+                        "mrcnn training")
+    args = parser.parse_args()
+    return args
+
+
+def structure(folder_list):
+    '''Create the structure for the project and downoald necessary file'''
+    for name in folder_list:
+        os.makedirs(name, exist_ok=True)
+
+
+def images_transfert(dataset, o_folder, f_folder):
+    '''Copy selected images from a folder to another one'''
+    filelist = list()
+    for image_name in dataset.ImageId.unique():
+        filelist.append(o_folder + image_name)
+    print(f"[INFO] Copying of {len(filelist)} images from {o_folder} to "\
+          f"{f_folder} on progress. Please wait.")
+    for file in filelist:
+        shutil.copy2(file, f_folder)
+    print("[INFO] Copy is done.")
+
+
+def images_copy(csv_file, o_folder, p_folder, percent_images=1, val_size=0.2):
+    '''Copy images for the training with respected proportion
+    dataset: A pandas dataset is provided,
+    percent_images: A percent of images to copy from Kaggle folder to project,
+    val_size: percent of val images,
+    o_folder: path to the origin folder,
+    p_folder: path to the project folder'''
+    dataset = pd.read_csv(csv_file)
+    dataset["Mask"] = dataset["EncodedPixels"].apply(
+        lambda x: 0 if isinstance(x, float) else 1)
+    X = dataset.iloc[:, :2]
```

Comment:
Deep learning tutorials and libraries taught to you that `X` is a good variable name in python, but it's not!!! No cheating allowed https://realpython.com/python-pep8/

Reply:
I would use `xs` in that case (in the sense of several `x`)